src/legacy/ui/public/courier/fetch/components/shard_failure_table.tsx

```diff
@@ -0,0 +1,116 @@
+/*
+ * Licensed to Elasticsearch B.V. under one or more contributor
+ * license agreements. See the NOTICE file distributed with
+ * this work for additional information regarding copyright
+ * ownership. Elasticsearch B.V. licenses this file to you under
+ * the Apache License, Version 2.0 (the "License"); you may
+ * not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *    http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing,
+ * software distributed under the License is distributed on an
+ * "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
+ * KIND, either express or implied.  See the License for the
+ * specific language governing permissions and limitations
+ * under the License.
+ */
+import React, { useState, ReactElement } from 'react';
+// @ts-ignore
+import { EuiInMemoryTable, EuiButtonIcon } from '@elastic/eui';
+// @ts-ignore
+import { RIGHT_ALIGNMENT } from '@elastic/eui/lib/services';
+import { i18n } from '@kbn/i18n';
+import { ShardFailureDescription } from './shard_failure_description';
+import { ShardFailure } from './shard_failure_types';
+
+export interface ListItem extends ShardFailure {
+  id: string;
+}
+
+export function ShardFailureTable({ failures }: { failures: ShardFailure[] }) {
+  const itemList = failures.map((failure, idx) => ({ ...{ id: String(idx) }, ...failure }));
+  const initalMap = {} as Record<string, ReactElement>;
+
+  const [expandMap, setExpandMap] = useState(initalMap);
+
+  const collapseLabel = i18n.translate(
+    'common.ui.courier.fetch.shardsFailedModal.tableRowCollapse',
+    {
+      defaultMessage: 'Collapse',
+      description: 'Collapse a row of a table with failures',
+    }
+  );
+
+  const expandLabel = i18n.translate('common.ui.courier.fetch.shardsFailedModal.tableRowExpand', {
+    defaultMessage: 'Expand',
+    description: 'Expand a row of a table with failures',
+  });
+
+  const columns = [
+    {
+      align: RIGHT_ALIGNMENT,
+      width: '40px',
+      isExpander: true,
+      render: (item: ListItem) => (
+        <EuiButtonIcon
+          onClick={() => {
+            // toggle displaying the expanded view of the given list item
+            const map = Object.assign({}, expandMap);
+            if (map[item.id]) {
+              delete map[item.id];
+            } else {
+              map[item.id] = <ShardFailureDescription {...item} />;
+            }
+            setExpandMap(map);
+          }}
+          aria-label={expandMap[item.id] ? collapseLabel : expandLabel}
+          iconType={expandMap[item.id] ? 'arrowUp' : 'arrowDown'}
+        />
+      ),
+    },
+    {
+      field: 'shard',
+      name: 'Shard',
```

Comment:
Hmm, so these are terms of the API, but hardcoded as table header, when we translate them, type might no match with the terms we display directly by the JSON keys ... wat meinste @timroes

Reply:
Good question. I would put them into `i18n.translate` calls. Most likely since they are technical terms we anyway don't translate them in most languages, since they are part of our "non translation glossary". But since it's the table header I think let's put them into i18n.translate calls even to the danger of them being different from what we show in the expanded details.

Reply:
been there, done that